Fixes pyprobml-review/pyprobml-review#14

nitish1295 opened the pull request:
| Before PR |![image](https://user-images.githubusercontent.com/21181046/163676458-e71b8b1d-1340-4a26-993f-0a3b1a5db8cc.png) |
| :-: | :-: |
| After PR | ![image](https://user-images.githubusercontent.com/21181046/163676453-88428162-8da2-428f-b31c-b3746e8a1fb0.png)
|

patel-zeel commented on gauss_imputation_known_params_demo.ipynb:
```diff
@@ -0,0 +1,310 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #24.        plt.ylabel("Dimension of Data")
</pre><p>Data -&gt; data</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/15/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

nitish1295 replied:
Done

patel-zeel commented on gauss_imputation_known_params_demo.ipynb:
```diff
@@ -0,0 +1,310 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #1.    gen_imputer_data()
</pre><p>Since both the x-axis and y-axis are fixed. Let us have y-label only for the first subplot and remove y-label and y-ticklabels from other two. You can specify these details via a dictionary input to the function.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/15/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

nitish1295 replied:
@patel-zeel So the x-axis will still be visible for the other 2 correct?

patel-zeel replied:
Yes

nitish1295 replied:
Done

patel-zeel commented on gauss_imputation_known_params_demo.ipynb:
```diff
@@ -0,0 +1,310 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #1.    from ipywidgets import interact
</pre><p>Nice!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/15/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

nipunbatra commented:
I haven't looked at the notebook. 

I have the following comments:

1. It is not clear to me what the labels for both x-axis and y-axis mean. The caption is also not helping me
2. The grid lines seem to dominate the visual appearance. Perhaps they can be reduced in width and alpha?
3. would it make sense to create a legend? see: https://stackoverflow.com/a/58485655/743775 for inspiration

nitish1295 commented:
@nipunbatra 
@patel-zeel 

## Checklist

- [ ] 1. It is not clear to me what the labels for both x-axis and y-axis mean. The caption is also not helping me
- [x] 2. The grid lines seem to dominate the visual appearance. Perhaps they can be reduced in width and alpha?
- [ ] 3. would it make sense to create a legend? see: https://stackoverflow.com/a/58485655/743775 for inspiration
- [x] 4. Removed y-axis labels and ticks for the second and third graph

## Key Points

1. I agree they do not convey much, I am still trying to find relevant captions and axes labels. For now, I have updated them with the Observed data matrix, True data matrix, and Imputed Data matrix which I think is better than before
2. I have updated the grid to be thinner, this can be readjusted based on your inputs
3. I have created a legend, although I did face the following issues while doing this:
    - Had to duplicate the code in `hinton_diagram` from `probml_utils` to get the size values. 
    - The current legend has a `plt.plot` to define squares. markers I did try using `plt.Rectangle` but was unable to adjust the size using that. 
    - Please note that code in function `plot_hinton_data` will be modified based once we finalize labels. You can uncomment the `plt.legend` statement to see the legend. 
4. I have hidden the y-axis label for now. I was unsure if I was required to remove x-axis labels as well . While using `set_ticklabels([])` there were alignment issues with the x-axis hence I have hidden them.


| Before PR |![image](https://user-images.githubusercontent.com/21181046/163676458-e71b8b1d-1340-4a26-993f-0a3b1a5db8cc.png) |
| :-: | :-: |
| 2 and 4 |![image](https://user-images.githubusercontent.com/21181046/164264879-a14b8a36-60aa-44da-942e-c86f3c6a1884.png)|
| 3 | ![image](https://user-images.githubusercontent.com/21181046/164243027-6854e4c9-5cf9-4caa-a1b3-e2cce068be7d.png)|

nitish1295 commented:
@nipunbatra 
@patel-zeel 

## Checklist

- [ ] 1. It is not clear to me what the labels for both x-axis and y-axis mean. The caption is also not helping me
- [x] 2. I think all the three figures should be on the same scale. That way, the observations, which are a subset of True data should have same patch sizes. Currently that is not the case. Or, am I missing something?
- [x] 3. The grid is still heavy..one way would be to have the patches lie higher on zorder compared to grid.
- [ ] ~~4. would it make sense to create a legend? see: https://stackoverflow.com/a/58485655/743775 for inspiration~~ Removing legend as discussed

## Key Points
1. The text leading up to the figure is as follows:
> Figure 3.7 shows a simple example. We sampled N vectors from a D = 10-dimensional Gaussian,
and then deliberately “hid” 50% of the data in each sample (row).

Based on the above text, I feel x-axis label **Number of Samples** does convey the required information. Kindly let me know if    you still think this must be modified to **Sample Number**. Please let me know if these captions are acceptable

2. You are correct the scale is not similar between graphs due to this line in probml-utils https://github.com/probml/probml-utils/blob/7e26e33c43f1a7a786f6f9742b984f111eaa4805/probml_utils/pyprobml_utils.py#L117. This line creates a max_weight value based on the data in the matrix which is used to scale down the area of the patches. Since the data is different in every matrix passed to the function, so is this scaling factor. Luckily, we can also pass this value as a parameter which is what I have done

Latexified Image after Updates:
![image](https://user-images.githubusercontent.com/21181046/165339061-9d03bd0d-639d-4f5b-b994-d67a0eebb70d.png)

nitish1295 commented:
- [x] Update x-axis label to Sample Number

![image](https://user-images.githubusercontent.com/21181046/165454264-35e094a5-d8e2-4c66-8083-6d83cf8c99ca.png)

Hi @karm-patel 
Pushing here since there were multiple changes going in the repositories. Just wanted to confirm if there is anything else that needs to be modified before pushing into the master repo. Till now I have accommodated the following
- [x] Updated code based on https://github.com/probml/pyprobml/pull/766
- [x] Updated code based on https://github.com/probml/probml-utils/pull/1

karm-patel commented on notebooks/book1/03/gauss_imputation_known_params_demo.ipynb:
```diff
@@ -0,0 +1,322 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #9.        from probml_utils import savefig, latexify, is_latexify_enabled
</pre><p>you can add just 1 line in try...except and then other2 lines can go after that.</p><p>like this,</p><p>try:</p><p>	from probml_utils import pyprobml_utils as pml</p><p>except:</p><p>	5pip install git+https:// github.com/probml/probml-utils.git</p><p>	from probml_utils import pyprobml_utils as pml</p><p> </p><p>from probml_utils import savefig, latexify, is_latexify_enabled</p><p>from probml_utils import gauss_utils as gauss</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/15/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

nitish1295 replied:
Oh yes, that is correct

nitish1295 replied:
Thanks for catching that. Fixed this, will the master repo
